earthaccess/store.py

```diff
@@ -709,6 +714,8 @@ def _download_file(self, url: str, directory: Path) -> str:
                 # of one per file, see #913
                 self._clone_session_in_local_thread(original_session)
                 session = self.thread_locals.local_thread_session
+                retries = Retry(total=retries)
+                session.mount("https://", HTTPAdapter(max_retries=retries))
```

Comment:
I suspect this is better suited for the code that initializes the session object earthaccess uses, but I'll look into that more closely later.